addons/analytic/models/analytic_plan.py

```diff
@@ -145,17 +149,30 @@ def _compute_analytic_account_count(self):
         for plan in self:
             plan.account_count = len(plan.account_ids)
 
+    def _compute_all_children_ids(self):
+        self.env.cr.execute("""
+            SELECT parent.id,
+                 array_agg(child.id) as children_ids
+            FROM account_analytic_plan parent
+            JOIN account_analytic_plan child ON child.parent_path LIKE parent.parent_path || %s
+            WHERE parent.id IN %s
+            GROUP BY parent.id
+        """, ('%', tuple(self.ids)))
+        for plan, res in zip(self, self.env.cr.dictfetchall()):
+            children_ids = [p for p in res.get('children_ids', []) if p != plan.id]
+            plan.all_children_ids = self.env['account.analytic.plan'].browse(children_ids)
+
     @api.depends('account_ids', 'children_ids')
     def _compute_all_analytic_account_count(self):
         plans_count = dict(
             self.env['account.analytic.account']._read_group(
-                domain=[('root_plan_id', 'in', self.ids)],
+                domain=[('plan_id', 'child_of', self.ids)],
                 aggregates=['id:count'],
-                groupby=['root_plan_id']
+                groupby=['plan_id']
             )
         )
         for plan in self:
-            plan.all_account_count = plans_count.get(plan, 0)
+            plan.all_account_count = sum(plans_count.get(p, 0) for p in (plan.all_children_ids + plan))
```

Comment:
Note that the query includes `plan` inside of `plan.all_children_ids`
```suggestion
            plan.all_account_count = sum(plans_count.get(child, 0) for child in plan.all_children_ids)
```